servicetalk-http-api/src/test/java/io/servicetalk/http/api/DefaultServiceDiscoveryRetryStrategyTest.java

```diff
@@ -175,6 +188,22 @@ void noRetainErrorWithNoAddresses() throws Exception {
         sendUpAndVerifyReceive(state, "addr1", sdEvents);
     }
 
+    @ParameterizedTest(name = "{displayName} [{index}]: retainAddressesTillSuccess={0}, available={1}")
+    @MethodSource("booleanMatrix")
+    void duplicatedAddresses(final boolean retainAddressesTillSuccess, final boolean available) throws Exception {
+        State state = new State(retainAddressesTillSuccess);
+        TestPublisher<Collection<ServiceDiscovererEvent<String>>> sdEvents = state.pubs.take();
+        final String addr = "addr1";
+        if (!available) {
+            // First, emit available event for the same address
+            sendUpAndVerifyReceive(state, addr, sdEvents);
+        }
+        final DefaultServiceDiscovererEvent<String> evt1 = new DefaultServiceDiscovererEvent<>(addr, available);
+        final DefaultServiceDiscovererEvent<String> evt2 = new DefaultServiceDiscovererEvent<>(addr, available);
```

Comment:
duplicate events (e.g. <addr, available>, <addr, available>) is not a valid use of SD API. can you clarify what this test is meant to verify?

Reply:
Let's consolidate discussion here: https://github.com/apple/servicetalk/pull/1586#discussion_r641021922